### PR TITLE
fix: send JWT when calling security API

### PR DESCRIPTION
**Pull request by lumaxis**
Add a JWT authorization header when calling the npm-internal security API, fixes https://github.com/github/npm/issues/658

### CODE REVIEW

File: test/spec/analyze/collect/npm.js [hunk 1]
```diff
@@ -46,6 +46,17 @@ describe('npm', () => {
         .get((path) => path.indexOf('/downloads/range/') === 0)
         .reply(404, { error: 'package cross-spawn not found' });
 
+        sepia.nock('http://127.0.0.1:5984')
```
Comment (lumaxis) [inline test/spec/analyze/collect/npm.js:49]: Fixed this one test, only to realize that this project is using `mocha`'s `--bail` option (which exits immediately on the first failing test) and there was a bunch more broken tests waiting for me 🙃

File: .travis.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ node_js:
 script: "npm run test-travis"
 before_install:
   - sudo apt-get install -y bsdtar
+  - npx makeshift -s @npm -r https://registry.internal.npmjs.com
```
Comment (gordey4doronin) [inline .travis.yml:8]: Why adding internal registry if we don't use it for any packages?

Comment (lumaxis) [reply .travis.yml:8]: We do, there is a `@npm/fetch-coverage` package

Comment (gordey4doronin) [reply .travis.yml:8]: Okay, so the CI was broken for a while.. 👌

File: test/spec/analyze/collect/npm.js [hunk 3]
```diff
@@ -46,6 +46,17 @@ describe('npm', () => {
...
+        .get('/npm/_design/app/_view/dependedUpon')
```
Comment (gordey4doronin) [inline test/spec/analyze/collect/npm.js:50]: Looks like missing indentation. But I guess you try to keep the existing style? 🤔

Comment (lumaxis) [reply test/spec/analyze/collect/npm.js:50]: I actually personally prefer this style 😜 And yes, here I just used the existing pattern 🙂

Comment (gordey4doronin) [reply test/spec/analyze/collect/npm.js:50]: 🙃

**Pull request merged**